### PR TITLE
Refactor: move migration script test out of linter

### PR DESCRIPTION
**Pull request by bershanskiy**
This PR is very similar to #9229, but it refactors a different function, `testMigrations()`.

This PR moves a unit test out of [linter](https://github.com/mdn/browser-compat-data/blob/master/test/lint.js#L161) and into a separate mocha test file similar to [`utils.test.js`](https://github.com/mdn/browser-compat-data/blob/master/test/utils.test.js). This unit test is still called by `npm run test`, since [`test`](https://github.com/mdn/browser-compat-data/blob/master/package.json#L52) script includes `npm run mocha`.

For reviewer convenience, I split it up into two commits: the actual change is in the first commit and the second commit is just `npm run fix`.

### Before:
```
$ npm run test

> @mdn/browser-compat-data@3.1.1 test
> npm run mocha && npm run lint


> @mdn/browser-compat-data@3.1.1 mocha
> mocha "test/**.test.js"



  utils
    √ `escapeInvisibles()` works correctly


  1 passing (3ms)


> @mdn/browser-compat-data@3.1.1 lint
> node test/lint
> [rest of output]
```

### After:
```
$ npm run test

> @mdn/browser-compat-data@3.1.1 test
> npm run mocha && npm run lint


> @mdn/browser-compat-data@3.1.1 mocha
> mocha "**/**.test.js"



  migration scripts
    √ `removeWebViewFlags()` works correctly

  utils
    √ `escapeInvisibles()` works correctly


  2 passing (6ms)


> @mdn/browser-compat-data@3.1.1 lint
> node test/lint
> [rest of output]
```

A checklist to help your pull request get merged faster:
- [ ] Summarize your changes
- [ ] Data: link to resources that verify support information (such as browser's docs, changelogs, source control, bug trackers, and tests)
- [ ] Data: if you tested something, describe how you tested with details like browser and version
- [ ] Review the results of the linter and fix problems reported (If you need help, please ask in a comment!)
- [ ] Link to related issues or pull requests, if any

### CODE REVIEW

**Comment by bershanskiy** [comment]
(Force-pushed after rebase.)

**Comment by bershanskiy** [comment]
Just to clarify: this PR is OK as is, but I marked it as "Draft" because it is bound to merge conflict with #9229.

**Comment by bershanskiy** [comment]
(Force-pushed after rebase.)

**Comment by bershanskiy** [comment]
I think this will no longer have merge conflicts with #9229, so marking this as ready.

File: scripts/migrations/002-remove-webview-flags.test.js [hunk 1]
```diff
@@ -131,29 +131,25 @@ const tests = [
...
+      let expected = JSON.stringify(tests[i]['output'], null, 2);
+      let output = JSON.stringify(
+        JSON.parse(JSON.stringify(tests[i]['input']), removeWebViewFlags),
+        null,
+        2,
+      );
 
-    if (output !== expected) {
-      logger.error(chalk`{red WebView flags aren't removed properly!}
+      if (output !== expected) {
```
Comment (queengooborg) [inline scripts/migrations/002-remove-webview-flags.test.js:145]: Could we opt for using `assert.deepStrictEqual(output, expected)` here?

**Review by queengooborg — CHANGES_REQUESTED** [review]
I'm definitely down for script tests being moved out of `lint.js` -- thank you!  Just got a little nit, but LGTM overall:

File: scripts/migrations/002-remove-webview-flags.test.js (hunk 1 above)
Comment (bershanskiy) [reply scripts/migrations/002-remove-webview-flags.test.js:145]: If we elect to use `assert.deepStrictEqual(output, expected)` we can probably remove the whole JSON normalization (parsing and then stringification of data). Should I do that?

Comment (queengooborg) [reply scripts/migrations/002-remove-webview-flags.test.js:145]: Yeah, that's actually a great point, we could simply write this as:

```js
for (const test of tests) {
  assert.deepStrictEqual(output, expected);
}
```

Comment (bershanskiy) [reply scripts/migrations/002-remove-webview-flags.test.js]: I used the following:
```
    for (const test of tests) {
      const expected = test.output;
      const output = JSON.parse(JSON.stringify(test.input), removeWebViewFlags);
      assert.deepStrictEqual(output, expected);
    }
```

**Review by queengooborg — APPROVED** [review]
LGTM, thank you!

**Comment by queengooborg** [comment]
Looks like your other PR conflicted with this one, so this PR needs a quick rebase!

**Review by queengooborg — APPROVED** [review]
LGTM, thank you!

**Pull request merged**